autogpt_platform/backend/backend/blocks/email_block.py

```diff
@@ -1,22 +1,53 @@
 import smtplib
 from email.mime.multipart import MIMEMultipart
 from email.mime.text import MIMEText
+from typing import Literal
 
-from pydantic import BaseModel, ConfigDict
+from pydantic import BaseModel, ConfigDict, SecretStr
 
 from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
-from backend.data.model import BlockSecret, SchemaField, SecretField
+from backend.data.model import (
+    CredentialsField,
+    CredentialsMetaInput,
+    SchemaField,
+    UserPasswordCredentials,
+)
+from backend.integrations.providers import ProviderName
 
+TEST_CREDENTIALS = UserPasswordCredentials(
+    id="01234567-89ab-cdef-0123-456789abcdef",
+    provider="smtp",
```

Comment:
ew

(not sure how to fix this tho)

Reply:
what don't you like about it?

Reply:
"smtp" is not a provider

Reply:
I meannnnnn yeah fair. Take a look at this ticket https://github.com/Significant-Gravitas/AutoGPT/issues/9273 and share your thoughts there. Toran and I discussed this pretty heavily. I'm going to resolve this but I understand where you're coming from